core/src/main/java/com/opensymphony/xwork2/interceptor/DefaultResourceIsolationPolicy.java

```diff
@@ -0,0 +1,55 @@
+package com.opensymphony.xwork2.interceptor;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class DefaultResourceIsolationPolicy implements ResourceIsolationPolicy {
+
+    private List<String> exemptedPaths = new ArrayList<String>();
+
+    public List<String> getExemptedPaths(){
+        return this.exemptedPaths;
+    }
+
+    public void setExemptedPaths(List<String> paths){
+        this.exemptedPaths = paths;
+    }
+
+    @Override
+    public boolean isRequestAllowed(HttpServletRequest request) {
+
+        String site = request.getHeader(SEC_FETCH_SITE_HEADER);
+
+        // Allow same-site and browser-initiated requests
+        if (SAME_ORIGIN.equals(site) || SAME_SITE.equals(site) || NONE.equals(site)) {
+            return true;
+        }
+
+        // Apply exemptions: paths/endpoints meant to be served cross-origin
+        if (this.exemptedPaths.contains(request.getContextPath())) {
+            return true;
+        }
+
+        // Allow requests from browsers which don't send Fetch Metadata
+        if (request.getHeader(SEC_FETCH_SITE_HEADER) == null){
+            return true;
+        }
```

Comment:
it's too late to make this check! we've already extracted this header in the `site` variable and although this implementation is null-safe, let's make this check up front and return true as soon as possible